valuetypes.go

```diff
@@ -514,3 +516,123 @@ func (d DateTime) Map(data interface{}, target interface{}) error {
 		return ErrTypeMismatch
 	}
 }
+
+// Duration schema type for duration as integer seconds or string on the form:
+//
+//     /[-+]? (\d+ d(ays?)?)? (\d+ h((ours?)?|r))? (\d+ m(in(untes?)?)?)?/
+//
+// This allows for a lot of human readable time durations, examples:
+//
+//    '31 days 2 hours 5 min'
+//    '5 min'
+//    '2 hours 5 min'
+//    '+ 3 minutes'
+//    '- 2 hr'
+//    '- 3 days'
+//    '1d2h3m'
+//    '   1  day  2 hour  1 minutes '
```

Comment:
I want something that can be expressed as a JSON schema... So a  string matching​ a regex or and int